elastic/integrations
# ES package log pipelines

Closes https://github.com/elastic/integrations/issues/4046

### Summary

Updates the log pipelines so we correctly ingest all elasticsearch log types.

Also adds a docker-compose with an elasticsearch service that can be started with `elastic-package service up`. The service is configured to generate logs out of the box, with an additional container continuously executing a script (`generate-logs.sh`) that sends requests to ES to generate log types not created ootb

### Testing
- build the elasticsearch package: `cd packages/elasticsearch && elastic-package build`
- start a stack: `elastic-package stack up -v -d --version 8.5.0-SNAPSHOT`
- start the elasticsearch service: `cd packages/elasticsearch && elastic-package service up -v`
- install the elasticsearch package
  - a script is available to automate the installation with pre-configured values `./packages/elasticsearch/_dev/deploy/docker/scripts/install-package.sh`,
  - otherwise install the package manually. The logs basepath is `/tmp/service_logs/`.
- connect to Kibana at https://localhost:5601 and verify all 5 log types are correctly ingested. One can lookup the `data_stream.dataset` for `logs-*` pattern in discover. Note that 4 log dataset should be `elasticsearch.*` while deprecation is the other way around `deprecation.elasticsearch`. Asked the ES team whether we should take the opportunity to reverse it in https://github.com/elastic/integrations/issues/4046#issuecomment-1222626213

These logs are also surfaced in the Stack Monitoring UI. You can connect a local kibana (from main) to the elastic-package stack ([howto](https://github.com/elastic/observability-dev/blob/8c6b377f5844053d2d202219679d94957a28cc00/docs/infra-obs-ui/stack-monitoring_integration-packages.md#connecting-a-local-kibana))
